### PR TITLE
JDBC - Fix missing metadata

### PR DESCRIPTION
**Pull request by Jens-H**
This PR repairs two meta data functions, that were not updated with the JDBC latest additions.

Also some test cases are added.

### CODE REVIEW

File: tools/jdbc/src/main/java/org/duckdb/DuckDBResultSetMetaData.java [hunk 1]
```diff
@@ -150,8 +154,21 @@ public String getColumnClassName(int column) throws SQLException {
 			return Date.class.toString();
 		case Types.TIMESTAMP:
 			return Timestamp.class.toString();
+		case Types.TIME_WITH_TIMEZONE:
+			return OffsetDateTime.class.toString();
 		case Types.BLOB:
 			return DuckDBBlobResult.class.toString();
+		case Types.DECIMAL:
+			switch (column_types_meta[column].type_size) {
```
Comment (Mytherin) [inline tools/jdbc/src/main/java/org/duckdb/DuckDBResultSetMetaData.java:162]: Should decimals return a `short` or `integer` class? Internally this is what they use but within Java not no?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good to me - one question.

File: tools/jdbc/src/main/java/org/duckdb/DuckDBResultSetMetaData.java (hunk 1 above)
Comment (Jens-H) [reply tools/jdbc/src/main/java/org/duckdb/DuckDBResultSetMetaData.java:162]: Yes, you are right. Thanks for catching this. It should always return BigDecimal.

**Review by Mytherin — APPROVED** [review]

**Comment by hannes** [comment]
Hey @Jens-H thanks for all your PRs on the JDBC client, could you ping me at `hannes@duckdblabs.com` please? Would love to have a short chat.

**Pull request merged**